test/run-drun/basic-cycles.mo

```diff
@@ -91,6 +92,7 @@ actor a {
   public func go() : async (){
     await overflow();
     await iter();
+    assert 1000 == burn<system> 1000
```

Comment:
Detail: Maybe we could test burn 0 and a very large `u128` just to check the double `u64` ordering?

Reply:
I can test 0, but the very large will clip to available...

Reply:
Yes, I thought so. Just an idea, not important.

Reply:
I think we are fine